trpcultivate_phenotypes/tests/src/Kernel/Services/Traits/ValidTraitTest.php

```diff
@@ -244,55 +244,164 @@ public function testTraitsServiceGetters() {
+      // should now return the units: UnitABCxxxxx and AnotherUnit.
+      [
+        'Trait Name' => 'Re-using Method',
+        'Trait Description' => 'This is re-using the method MethodABC',
+        'Method Short Name' => $method_short_name,    // Same as above.
+        'Collection Method' => $method_of_collection, // Same as above.
+        'Unit' => 'AnotherUnit',
+        'Type' => 'Quantitative'
+      ]
     ];
+    
+    // This will create the following traits:
+    // TraitABC 
+    //   description: TraitABC Description,
+    //   method short name: MethodABC-SName
+    //   collection method: MethodABC -  Pull from ground
+    //   unit: UnitABC
+    //   type: Quantitative
+    // 
+    // Re-using Method 
+    //   description: This is re-using the method MethodABC,
+    //   method short name: MethodABC-SName
+    //   collection method: MethodABC -  Pull from ground
+    //   unit: AnotherUnit
+    //   type: Quantitative
 
-    // Set genus to use by the traits service.
-    $this->service_traits->setTraitGenus($this->genus);
-    // Save the trait.
-    $trait_assets = $this->service_traits->insertTrait($trait);
-
-    // Test get trait.
 
-    $trait_id = $trait_assets['trait'];
-    $trait_name = $trait['Trait Name'];
-
-    // Get trait by id.
-    $t = $this->service_traits->getTrait(['id' => $trait_id]);
-    $this->assertEquals($trait_name, $t->name, 'Trait not found (by trait id).');
+    // Set genus to use by the traits service.
+    $this->service_traits->setTraitGenus($this->genus);    
+    $method_id = 0;
 
-    // Get trait by name.
-    $t = $this->service_traits->getTrait(['name' => $trait_name]);
-    $this->assertEquals($trait_name, $t->name, 'Trait not found (by trait name).');
+    // Save the trait.
+    foreach($traits as $i => $trait) {
+      $trait_assets = $this->service_traits->insertTrait($trait);
+      
+      // Test trait service.
+      $trait_id = $trait_assets['trait'];
+      $trait_name = $trait['Trait Name'];
+  
+      // Get trait by id.
+      $t = $this->service_traits->getTrait(['id' => $trait_id]);
+      $this->assertEquals($trait_name, $t->name, 'Trait not found (by trait id).');
+
+      // Get trait by name.
+      $t = $this->service_traits->getTrait(['name' => $trait_name]);
+      $this->assertEquals($trait_name, $t->name, 'Trait not found (by trait name).');
+
+      // Test get trait method.
+      $method_name = $trait['Method Short Name'];
+
+      // Get trait method by trait id.
+      $m = $this->service_traits->getTraitMethod(['id' => $trait_id]);
+      $this->assertEquals($method_name, $m[0]->name, 'Trait method not found (by trait id).');
+
+      // Get trait method by trait name.
+      $m = $this->service_traits->getTraitMethod(['name' => $trait_name]);
+      $this->assertEquals($method_name, $m[0]->name, 'Trait method not found (by trait name).');
+
+      // Units - UnitABC and AnotherUnit.
+      $method_id = $method_id = $m[0]->cvterm_id;
```

Comment:
Variable setting seems a bit excessive here 😅 . Also, does `$method_id` need to be set on line 291 in this case?

Reply:
That is a good catch, in both cases